### ISSUE TITLE
Loading Texture 2D with NDI source from preset doesn't work

### ISSUE THREAD

**Issue by TimPietrusky** [vcync/modV#537]
**Operating system and application version**
 - OS: Windows
 - Version: next

**Describe the bug**
When trying to  load a preset that contains a `Texture 2D` which loads a canvas which loads an NDI source, every module in the preset is broken.

**To Reproduce**
1. Use an NDI source via a Texture 2D
2. Save the preset
3. Reload modV
4. Add the same NDI source again
5. Load the preset
6. See that the module can't be loaded and these errors in the console

```
Uncaught TypeError: Cannot read property 'context' of undefined at modules/outputs.js
```

Especially this line is the problem: `auxillaryCanvas: state => id => state.auxillary[id].context.canvas`

**Expected behavior**
The preset and all of its modules can be loaded and I see an error for the  Texture 2D that the NDI source is not there anymore.

**Comment by TimPietrusky** [comment]
I guess as this is only using the ID which is not unique, the same NDI source can't be found, even when the same source is added again as modV is using it's own uuid to make the connection.

**Comment by 2xAA** [comment]
Yeah, we're tied to Grandiose here. The receivers need to be created for the state to be in sync with the loaded presets, but we don't know which sources are available and potentially never will know as the source discovery of Grandiose isn't exactly reliable.

Maybe a solution here is to either ask the user to add NDI sources back manually with a dialog and/or substitute missing NDI sources with a placeholder texture?

**Comment by 2xAA** [comment]
Another solution here would be to save the name, start a source scan and if the names match update the module property which is looking for the NDI source.

Names aren't guaranteed to be unique, but better than nothing?
How many NDI sources are you realistically going to have on your network?
